### PR TITLE
chore: Convert subcategory in config to PascalCase

### PR DESCRIPTION
**Pull request by cshfang**
#### Description of changes

This PR updates `inAppMessaging` -> `InAppMessaging` and cleans up some of the configuration around the category.

#### Checklist

- [x] PR description included
- [x] `yarn test` passes
- [ ] Tests are [changed or added](https://github.com/aws-amplify/amplify-js/blob/main/CONTRIBUTING.md#steps-towards-contributions)
- [ ] Relevant documentation is changed or added (and PR referenced)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/notifications/src/Notifications.ts [hunk 1]
```diff
@@ -41,20 +37,15 @@ class NotificationsClass {
...
+	configure = ({ Notifications: config }: NotificationsConfig = {}) => {
+		this.config = { ...this.config, ...config };
 
 		logger.debug('configure Notifications', config);
 
-		const inAppMessagingConfig = this.inAppMessaging.configure(
-			this.config.inAppMessaging
-		);
+		// Configure sub-categories
+		this.inAppMessaging.configure(this.config.InAppMessages);
```
Comment (calebpollman) [inline packages/notifications/src/Notifications.ts:46]: Maybe I'm missing something but should this line be:
```
this.inAppMessaging.configure(this.config.InAppMessaging);
```

Or are we re-naming the config key to `InAppMessages`?

Comment (elorzafe) [inline packages/notifications/src/Notifications.ts:46]: Need optional chaining?

Comment (cshfang) [reply packages/notifications/src/Notifications.ts:46]: `this.config` will always be defined as an object even if it somehow managed to spread nothing so it should be a safe access here.

Comment (cshfang) [reply packages/notifications/src/Notifications.ts:46]: Doh good catch!

**Review by calebpollman — APPROVED** [review]
🚢 🚢 🚢

**Review by elorzafe — APPROVED** [review]
LGTM 🌮

**Pull request merged**